src/providers/validationProvider.ts

```diff
@@ -32,8 +33,21 @@ export async function doValidate(
       validationManager.getValidationFromCache(textDocument.uri) ||
       new Map<string, Diagnostic[]>();
   } else {
-    // full validation with ansible-lint
-    diagnosticsByFile = await context.ansibleLint.doValidate(textDocument);
+    // full validation with ansible-lint or ansible syntax-check (if ansible-lint is not installed or disabled)
+
+    const lintAvailability = libraryChecker('ansible-lint', '--version');
+
+    if (await lintAvailability) {
+      console.debug('Validating using ansible-lint');
+      diagnosticsByFile = await context.ansibleLint.doValidate(textDocument);
+    }
+
+    if (!diagnosticsByFile || !(await lintAvailability))
+      console.debug('Validating using ansible syntax-check');
+    diagnosticsByFile = await context.ansibleSyntaxCheck.doValidate(
+      textDocument
+    );
```

Comment:
the `if` statement does not cover `diagnosticsByFile` assignment, hence it will always run, and unexpectedly replace existing diagnostics from `ansible-lint`.

Reply:
> Good work, but there are some issues and copy-paste left-overs.
> 
> The `libraryChecker` name sounds misleading to me, given what it does and what it works with, but maybe that's just me.

Hey @tomaciazek, we are discussing about a better name that could go in place of `libraryChecker` and once we finalize it, I will update it accordinly.
Thanks.

Reply:
I refactored it to `getExecutablePatch` (which is similar to shutil.which() in python), that returns the path to the executable if found, else returns undefined